shared/core/types/transaction.go

```diff
@@ -2,48 +2,49 @@ package types
 
 import (
 	"bytes"
+	"fmt"
 
 	"github.com/pokt-network/pocket/shared/codec"
 	"github.com/pokt-network/pocket/shared/crypto"
+	"google.golang.org/protobuf/proto"
 )
 
-func TxFromBytes(txProtoBytes []byte) (*Transaction, Error) {
+var _ ITransaction = &Transaction{}
+
+// TxFromBytes unmarshals a proto serialized Transaction to a Transaction protobuf
+func TxFromBytes(txProtoBz []byte) (*Transaction, error) {
 	tx := &Transaction{}
-	if err := codec.GetCodec().Unmarshal(txProtoBytes, tx); err != nil {
-		return nil, ErrUnmarshalTransaction(err)
+	if err := codec.GetCodec().Unmarshal(txProtoBz, tx); err != nil {
+		return nil, err
 	}
 	return tx, nil
 }
 
-func TxHash(txProtoBytes []byte) string {
-	return crypto.GetHashStringFromBytes(txProtoBytes)
+// TxHash returns the hash of the proto marshaled transaction
+func TxHash(txProtoBz []byte) string {
+	return crypto.GetHashStringFromBytes(txProtoBz)
 }
 
-var (
-	_ Validatable  = &Transaction{}
-	_ ITransaction = &Transaction{}
-)
-
 // `ITransaction` is an interface that helps capture the functions added to the `Transaction` data structure.
 // It is unlikely for there to be multiple implementations of this interface in prod.
 type ITransaction interface {
-	GetMessage() (Message, Error)
-	Sign(privateKey crypto.PrivateKey) Error
-	Hash() (string, Error)
+	GetMessage() (proto.Message, error)
+	Sign(privateKey crypto.PrivateKey) error
+	Hash() (string, error)
 	SignableBytes() ([]byte, error)
 	Bytes() ([]byte, error)
 	Equals(tx2 ITransaction) bool
 }
 
-func (tx *Transaction) ValidateBasic() Error {
+func (tx *Transaction) ValidateBasic() error {
 	// Nonce cannot be empty to avoid transaction replays
 	if tx.Nonce == "" {
-		return ErrEmptyNonce()
+		return fmt.Errorf("nonce cannot be empty")
```

Comment:
Any particular reason why you want to move away from the typed error pattern that used the `Error` type?

If we are moving away from it (like it seems given your changes) I guess we should axe it all together otherwise it's just unused code. 🤔

Generally speaking I think we should instead embrace the usage of the pattern across the whole codebase